rviz_default_plugins/src/rviz_default_plugins/displays/image/ros_image_texture.cpp

```diff
@@ -0,0 +1,315 @@
+{
+  while (buffer.size() > median_frames_ - 1) {
+    buffer.pop_back();
+  }
+  buffer.push_front(value);
+}
+
+double ROSImageTexture::computeMedianOfBuffer(const std::deque<double> & buffer) const
+{
+  std::deque<double> buffer2 = buffer;
+  nth_element(buffer2.begin(), buffer2.begin() + buffer2.size() / 2, buffer2.end());
+  return *(buffer2.begin() + buffer2.size() / 2);
+}
+
+template<typename T>
+std::vector<uint8_t>
+ROSImageTexture::createNewNormalizedBuffer(
+  const T * image_data,
+  size_t image_data_size,
+  T minValue,
+  T maxValue) const
+{
+  std::vector<uint8_t> buffer;
+  buffer.resize(image_data_size, 0);
+  uint8_t * output_ptr = &buffer[0];
+
+  // Rescale floating point image and convert it to 8-bit
+  double range = maxValue - minValue;
+  if (range > 0.0) {
+    const T * input_ptr = image_data;
+
+    // Rescale and quantize
+    for (size_t i = 0; i < image_data_size; ++i, ++output_ptr, ++input_ptr) {
+      double val = (static_cast<double>(*input_ptr - minValue) / range);
+      if (val < 0) {val = 0;}
+      if (val > 1) {val = 1;}
+      *output_ptr = val * 255u;
+    }
+  }
+  return buffer;
+}
+
+ImageData::ImageData(std::string encoding, const uint8_t * data_ptr, size_t size)
+: encoding_(std::move(encoding)),
+  pixel_format_(Ogre::PixelFormat::PF_R8G8B8),
+  data_ptr_(data_ptr),
+  size_(size)
+{
+}
+
+bool ROSImageTexture::update()
+{
+  sensor_msgs::msg::Image::ConstSharedPtr image;
+  bool has_new_image = fillWithCurrentImage(image);
+
+  if (!image || !has_new_image) {
+    return false;
+  }
+
+  new_image_ = false;
+
+  if (image->data.empty()) {
+    return false;
+  }
+
+  ImageData image_data = setFormatAndNormalizeDataIfNecessary(
+    ImageData(image->encoding, image->data.data(), image->data.size()));
+
+  width_ = image->width;
+  height_ = image->height;
+
+  Ogre::Image ogre_image;
+  try {
+    loadImageToOgreImage(image_data, ogre_image);
+  } catch (Ogre::Exception & e) {
+    RVIZ_COMMON_LOG_ERROR_STREAM("Error loading image: " << e.what());
+    return false;
+  }
+
+  texture_->unload();
+  texture_->loadImage(ogre_image);
+
+  return true;
+}
+
+bool ROSImageTexture::fillWithCurrentImage(sensor_msgs::msg::Image::ConstSharedPtr & image)
+{
+  std::lock_guard<std::mutex> lock(mutex_);
+
+  image = current_image_;
+  return new_image_;
+}
+
+ImageData ROSImageTexture::setFormatAndNormalizeDataIfNecessary(ImageData image_data)
+{
+  /* *INDENT-OFF* */  // Uncrustify is unable to handle the else if formatting
+  if (image_data.encoding_ == sensor_msgs::image_encodings::RGB8) {
+    image_data.pixel_format_ = Ogre::PF_BYTE_RGB;
+  } else if (image_data.encoding_ == sensor_msgs::image_encodings::RGBA8) {
+    image_data.pixel_format_ = Ogre::PF_BYTE_RGBA;
+  } else if (image_data.encoding_ == sensor_msgs::image_encodings::TYPE_8UC4 ||
+    image_data.encoding_ == sensor_msgs::image_encodings::TYPE_8SC4 ||
+    image_data.encoding_ == sensor_msgs::image_encodings::BGRA8) {
```

Comment:
If you need to turn off uncrustify, then please change this to be like this:

```c++
if (condition) {
  // ...
} else if (
  condition1 ||
  condition2 ||
  conditionN)
{
  // like this ^
  // The important thing is that this line and the line before are not
  // the same indentation...
} else if (condition1 ||
  condition2 ||
  conditionN) {
  // not like this ^
  // cannot easily see where the condition block ends and where the elseif block starts
  // also, if you wrap one part of the conditionals, wrap them all
} else {
  // ..
}
```

In fact, I think that if you do as I suggest above, uncrustify will not complain or change it.

Reply:
Yes, uncrustify would be happy... not so cpplint. 
That one enforces that closing and opening brackets should be on the same line. I'll disable cpplint for these lines adding a comment.